MyFirstSlideshow/app/build.gradle

```diff
@@ -20,4 +20,5 @@ android {
 
 dependencies {
     compile 'com.android.support:appcompat-v7:25.3.1'
+    compile 'com.karumi:dexter:4.1.0'
```

Comment:
Build.gradle - 23 - there should be implmentation in place of compile with * at the end so that any new update comes in it easly incorporate.

Reply:
Build.gradle - 22, 23 - use implementation instead of compile.